apps/passport-client/components/screens/AddSubscriptionScreen.tsx

```diff
@@ -138,28 +211,57 @@ export function SubscriptionInfoRow({
 function SubscribeSection({
   subscriptions,
   providerUrl,
+  providerName,
   info
 }: {
   subscriptions: FeedSubscriptionManager;
   providerUrl: string;
+  providerName: string;
   info: Feed;
 }) {
   const identity = useIdentity();
-  const [credential, setCredential] = useState<SerializedPCD | undefined>();
-
   const pcds = usePCDCollection();
-  console.log("PCDS", pcds);
+  const [subscribing, setSubscribing] = useState<boolean>(false);
 
-  const onSubscribeClick = useCallback(() => {
-    if (!confirm("are you sure you want to subscribe to this feed?")) return;
+  // Only two types of credential can be requested
+  // If none is specified, default to SemaphoreSignaturePCD
+  const requiredCredentialType =
+    info.credentialType === EmailPCDPackage.name
+      ? EmailPCDPackage.name
+      : SemaphoreSignaturePCDPackage.name;
 
-    subscriptions.subscribe(providerUrl, info, credential);
-  }, [subscriptions, providerUrl, info, credential]);
+  // If feed asked for email PCD, make sure we have one
+  const missingCredential =
+    requiredCredentialType === EmailPCDPackage.name &&
+    pcds.getPCDsByType(requiredCredentialType).length === 0;
 
-  const addCredential = useCallback(() => {
+  const onSubscribeClick = useCallback(() => {
     (async () => {
-      setCredential(
-        await SemaphoreSignaturePCDPackage.serialize(
+      if (!subscriptions.getProvider(providerUrl)) {
+        subscriptions.addProvider(providerUrl, providerName);
+      }
+
+      // If we have a credential type specified, and it's not a
+      // SemaphoreSignaturePCD, then look it up from PCDCollection.
+      // We can't do this with SemaphoreSignaturePCD because it's generated
+      // dynamically rather than stored in the collection.
```

Comment:
we should probably cache it in localstorage

Reply:
#639